litestar/_openapi/schema_generation/schema.py

```diff
@@ -142,22 +142,22 @@
 }
 
 
-def _get_type_schema_name(value: Any) -> str:
+def _get_type_schema_name(field_definition: FieldDefinition) -> str:
     """Extract the schema name from a data container.
 
     Args:
-        value: A data container
+        field_definition: A field definition instance.
 
     Returns:
         A string
     """
 
-    if name := getattr(value, "__schema_name__", None):
+    if name := getattr(field_definition.annotation, "__schema_name__", None):
         return cast("str", name)
 
-    name = get_name(value)
-    if args := get_args(value):
-        inner_parts = ", ".join(_get_type_schema_name(a) for a in args)
+    name = get_name(field_definition.annotation)
+    if field_definition.inner_types:
+        inner_parts = ", ".join(_get_type_schema_name(t) for t in field_definition.inner_types)
```

Comment:
This means we don't need to call `get_args()` again for the type and each of its inner types.